lib/checker.rb

```diff
@@ -6,8 +6,9 @@ def check(result, example)
     end
 
     def check_equal(result, records)
+      memory = result.fetch(:memory, {})
       records.each do |record, expected|
-        actual = result[:records][record]
+        actual = result[:records].merge(memory).fetch(record, '0000')
```

Comment:
This is quite ugly, the `fetch` is heavily dependant on the current output QSim has. We should have our own internal representation of the QSim machine and use it across all the component.

You have already solved the parsing issue, now please reify the machine so we don't have to do these kind of things everytime we want to get the value of a memory cell.